src/server/set_family.cc

```diff
@@ -1435,18 +1435,36 @@ void SScan(CmdArgList args, const CommandContext& cmd_cntx) {
 
 // Syntax: saddex key ttl_sec member [member...]
 void SAddEx(CmdArgList args, const CommandContext& cmd_cntx) {
-  string_view key = ArgS(args, 0);
-  string_view ttl_str = ArgS(args, 1);
+  CmdArgParser parser(args);
+
+  std::string_view key;
   uint32_t ttl_sec;
-  constexpr uint32_t kMaxTtl = (1UL << 26);
+  std::tie(key, ttl_sec) = parser.Next<std::string_view, uint32_t>();
 
-  if (!absl::SimpleAtoi(ttl_str, &ttl_sec) || ttl_sec == 0 || ttl_sec > kMaxTtl) {
+  if (auto err = parser.Error(); err) {
+    return cmd_cntx.rb->SendError(err->MakeReply());
+  }
+  constexpr uint32_t kMaxTtl = (1UL << 26);
+  if (ttl_sec == 0 || ttl_sec > kMaxTtl) {
     return cmd_cntx.rb->SendError(kInvalidIntErr);
   }
 
-  auto vals = args.subspan(2);
+  absl::flat_hash_set<std::string_view> vals;
+  vals.reserve(parser.Tail().size());
+  bool keepttl = false;
+
+  while (parser.HasNext()) {
+    auto val = parser.Next<std::string_view>();
+    if (!keepttl && absl::EqualsIgnoreCase(val, "KEEPTTL")) {
+      keepttl = true;
+      continue;
+    }
+
+    vals.insert(val);
+  }
+
```

Comment:
need to add validation here that at least one member is provided after skipping `KEEPTTL`

Reply:
added